generic: 6.1: backport support for generic spi-nor from SFDP data

Ansuel opened the pull request:
Backport patches for support of generic spi-nor from SFDP data for
kernel 6.1.

Kernel 5.15 have major rework of the info flags and it's not trustable
to backport this amount of changes and expect correct function of it.

All affected patches automatically refreshed using make
target/linux/refresh.

hauke commented:
nice. Should most of our flash chips then work without special patches?

When we merge this should we remove the patches adding additional flash chips and hope they will work without patches now? If they do not work people could still complain.

Ansuel commented:
depends if they have sfpd table... but that require manual check on a running device.

robimarko commented:
I somehow missed upstream finally doing this, but a lot of devices will still require SFPD fixups as there are plenty of broken SFPD-s

981213 commented:
Typo in the commit message: It's Serial Flash Discoverable Parameter or SFDP for short :)

Ansuel committed:
generic: 6.1: backport support for generic spi-nor from SFDP data

Backport patches for support of generic spi-nor from SFDP data for
kernel 6.1.

Kernel 5.15 have major rework of the info flags and it's not trustable
to backport this amount of changes and expect correct function of it.

All affected patches automatically refreshed using make
target/linux/refresh.

Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

Ansuel commented:
Sadly I had to drop 5.15 support for this as they made major rework for the info flags and I don't want to backport an hack patch for it... Better be safe than sorry... Also in theory 5.15 will be dropped so no reason to care about adding this feature also there.

hauke commented:
This looks good. Dropping kernel 5.15 support is no problem from my point of view.

Ansuel commented:
@hauke the guy also reported this to work correctly (aside from having the manu name dropped)
